### DISCUSSION TITLE
progress bar?

### DISCUSSION THREAD

**Discussion by zaourzag** [qgustavor/mega#95] [Q&A] [answered]
is there a progress bar?

**Comment by qgustavor** [comment, marked as answer]
You can implement a progress bar by listening to `progress` events on the stream returned by `.download()`. For reference implementations check [direct-mega's implementation](https://github.com/qgustavor/direct-mega/blob/87302d5e2a60a08570694f2a29e8f984bc1a7d5c/src/fallback.js#L121-L135) for a browser implementation and [megajs-cli](https://github.com/qgustavor/megajs-cli/blob/5a32b4a53d2317a12a14edd49da921a38f36b6e7/lib/util.js#L154-L182) for a Node implementation. Notice that in those implementations only a percentage is displayed to the user, since it's easier to implement, in order to implement a progress bar just feed this data to a [`<progress>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/progress) element (in browser) or a progress bar library such as [this one for Node](https://www.npmjs.com/package/cli-progress) and [this one for Deno](https://github.com/deno-library/progress).

**Reply by zaourzag** [reply]
What about adding smth that would make it easier to do?

**Reply by qgustavor** [reply]
No, as you can see it's not that hard to implement now and adding more code will require platform specific code. You didn't even specific what platform you're using, but, anyway there are many progress bar implementations. It's better to be library to only do one thing and do it well, which is handling MEGA (or at least don't extend the project scope too much to avoid feature creep; you can even say that the `progress` event is already getting out of scope as you could use other library to track progress, some even track speed and generate ETA for you).

**Reply by qgustavor** [reply]
Well, in the other hand, anyone is free to write a wrapper for this library that handle that.
